Source/Core/Section.swift

```diff
@@ -211,12 +211,29 @@ extension Section : MutableCollection, BidirectionalCollection {
             }
             return kvoWrapper.rows[position] as! BaseRow
         }
-        set { kvoWrapper.rows[position] = newValue }
+        set {
+            kvoWrapper.rows[position] = newValue
+            if position < kvoWrapper._allRows.count {
+                // Remove the previous row from the form
+                kvoWrapper._allRows[position].willBeRemovedFromForm()
```

Comment:
Yes I think so, why is the question? Rows are being indexed in the same way in both properties `rows` and `_allRows`, right?